jax/_src/api.py

```diff
@@ -1595,8 +1595,16 @@ def _get_axis_size(name: str, shape: Tuple[int, ...], axis: int):
     if kws:
       # if keyword arguments are included in the tree, we make adapt the error
       # message only to be about the positional arguments
+      full_tree = tree
       tree, leaf = treedef_children(tree)
-      assert treedef_is_leaf(leaf)
+
+      # Separately ensure that keyword arguments match in size.
+      if not treedef_is_leaf(leaf):
+        sizes = [x.shape[d] if d is not None else None for x, d in zip(vals, dims)]
+        sizes = tree_unflatten(full_tree, sizes)
+        print(repr(msg.format(f"the tree of axis sizes is:\n{sizes}")))
+        raise ValueError(msg.format(f"the tree of axis sizes is:\n{sizes}")) from None
```

Comment:
Should this `raise ... from e` like the error above?

Reply:
I did this to be consistent with the existing `raise ... from None` used further below.